code/modules/antagonists/heretic/transmutation_rune.dm

```diff
@@ -129,6 +132,36 @@
 			else
 				requirements_list[req_type]--
 
+		//If we have anything in optional_list
+		if(length(optional_list))
+			// Go through all of our optional atoms
+			for(var/optional_type in optional_list)
+				// We already have enough of this type, skip
+				if(optional_list [optional_type] <= 0)
+					continue
+				// If optional_type is a list of types, check all of them for one match.
+				if(islist(optional_type))
+					if(!(is_type_in_list(nearby_atom, optional_type)))
+						continue
+				else if(!istype(nearby_atom, optional_type))
+					continue
+				// if list has items, check if the strict type is banned.
+				if(length(banned_atom_types))
+					if(nearby_atom.type in banned_atom_types)
+						continue
+				// This item is a valid type. Add it to our selected atoms list.
+				selected_atoms |= nearby_atom
+				// Since its an optional atom, also add it to our optional_selected_atoms list.
+				optional_selected_atoms |= nearby_atom
+				// If it's a stack, we gotta see if it has more than one inside,
+				// as our requirements may want more than one item of a stack
+				if(isstack(nearby_atom))
+					var/obj/item/stack/picked_stack = nearby_atom
+					optional_list [optional_type] -= picked_stack.amount // Can go negative, but doesn't matter. Negative = fulfilled
+				// Otherwise, just add the mark down the item as fulfilled x1
+				else
+					optional_list [optional_type]--
```

Comment:
ok so these comments make me uncomfortable but I'm more unsure of what this code is actually for. 
also no spaces before []